### ISSUE TITLE
[Bug Report][3.2.1] VDataTable headers prop rowspan error

### ISSUE THREAD

**Issue by webdevnerdstuff** [vuetifyjs/vuetify#17241]
### Environment
**Vuetify Version:** 3.2.1
**Vue Version:** 3.2.47
**Browsers:** Chrome 112.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
1. Add rowspan to any of the headers array objects.
2. Notice there's an error in the console.

### Expected Behavior
Should change the rowspan.

### Actual Behavior
Causes an error.

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqllltvmzAUgP+KhTRlkxIgENKGZVOndpeue9jrVPpgwEm9go1sJ1UW5b/vmHuBlUhTpSr2uX3n4pzcH41PWWbud8TwjbUiaZZgRT4GDKH1foazLP+YH1JMWXnSxxgrPFM4TEjAAlX8+Y8Ex0TID4FRfgqMlpSCfy2LiZREKBBW/tDaanusgsJlHVUfCp611eKEo4wEzRSSRO20mKYZFwodkSAbdEIbwVM0gQQnWjniTCpUwqEP6F67PhbhFFUJ8dHkpsBDb+e2vQW3Yk/Z9t1kWmjhhG4ZaEmFhaouJUTU4D7a4ESS8vaJHECR4ZRUeoI/ywyDuZNfnPL/xzryNU64oEROpnUYwmI4FZ6iStwz/IIBVzMO2m2w6ptcYxHKV4wiLe+b/RRcEcpeMcwKjb7preBg9+ZfdhTEhdHDe21YtKqalapXdbcQ0oXVuQv+hzD0i293TUPAvCyWj+beqr6FUvhoadotNUgT+rGob0p+Hy1aahoOYs0r/0VufZjvJEkOKCSYDZG4F95LErtPshogaauVJPYYyR1Vd9D3AQpvfvmSwhkoyLIhrTEuuhDLMYjPUYKpGIJwlsUTqCHmQ11x+xBttRLjYgzjKzxgIkIBz36wLd5ynGXRDFHN4prNZTUhozW5jQiKACVFErP4mUaPg/Vxm2rnTKs+UkulNbRNzc4d2h88SWjGs8HarDp9ss3mohrZZp6ake1SOGMU17sswk/1V+ULCLvzblyzSb0a2POKcTmG8Y0zcoh4Gg6BLOzO03H71bgcAFmaTQIlyMIbI7nhbDf4gBdepyeO1x8Obz5Uj968Op2+6O/etVWsVFimxtQo9iks4sz8LTmD3wk5aFAKYIv7FXpgwJ7VZ9j/SmXSt6woZmAWk4TuhcmIsliWWlegZokdUzQls5inV67pmvYsJAqbjhVTqdpSk8h0FurVSQT4CowSOI+m6OZwRsRSEwI55ryIUF7NEhxKHaLn2tKrn4iZgC1FBBHnJtYxG0iuo9FLMP9tELATVD+h7El2Ch/JvOjlPvzPzHNneed1wNPUaOMCQG5nPPwFK7X1RA==)

### Other comments
Using the `rowspan` prop referenced here: https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/labs/VDataTable/types.ts#L12

Console Error:
```
caught (in promise) TypeError: Cannot read properties of undefined (reading 'push')
    at headers.ts:74:22
    at Array.forEach (<anonymous>)
    at watch.deep (headers.ts:71:10)
    at callWithErrorHandling (runtime-core.esm-bundler.js:154:18)
    at callWithAsyncErrorHandling (runtime-core.esm-bundler.js:162:17)
    at job (runtime-core.esm-bundler.js:1772:9)
    at flushPreFlushCbs (runtime-core.esm-bundler.js:309:7)
    at updateComponentPreRender (runtime-core.esm-bundler.js:5562:5)
    at ReactiveEffect.componentUpdateFn [as fn] (runtime-core.esm-bundler.js:5479:11)
    at ReactiveEffect.run (reactivity.esm-bundler.js:184:19)
(
```

**Comment by nekosaur** [comment]
What kind of header structure are you looking to create? There's not much point in using rowspan=2 when you only have one row.

**Comment by webdevnerdstuff** [comment]
@nekosaur tbh, I didn't even know that the v2 data-table had this prop as I've never personally used it before. So when I saw it in the v3 source code I'd test it out since I was testing some of the others. I'm building a drilldown table component for work, so I've just been testing everything I can think of that's available.  I agree that there's not much a point in using rowspan=2.

**Comment by nekosaur** [comment]
I'll close this since there's nothing actionable here. I don't have much time to look at that, but might I suggest you add a demo link that shows the features of it.

**Comment by webdevnerdstuff** [comment]
@nekosaur I think the only actionable thing to do would be to just remove it from the `DataTableHeader` type, which might avoid confusion to anybody else that might be looking deeper into the types.

I did test it in the v2 version and it did work, but like you said above, there's not much a point in using it even if it did work. The only reason I tested it was because I found it in your types.

https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/labs/VDataTable/types.ts#L14
